### PR TITLE
Fixed #306

### LINKED ISSUE #306: Linux: submenu detaches from wcm after alt-tab
Here is an example http://i57.tinypic.com/5phd1t.png

### PR DESCRIPTION
**Pull request by helgp**
See individual commit messages for details.

Source of the bug was in original Wal Commander code.

### CODE REVIEW

File: src/swl/swl_wincore.cpp [hunk 1]
```diff
@@ -465,13 +465,10 @@ namespace wal
...
-		for ( int i = 0; i < childList.count(); i++ )
-			if ( childList[i]->type & Win::WT_POPUP )
-			{
-				childList[i]->PopupTreeList( list );
-			}
+		if (type & Win::WT_POPUP)
+			list.append(handle);
+		for (int i = 0; i < childList.count(); i++)
+			childList[i]->PopupTreeList(list);
 	}
```
Comment (corporateshark) [inline src/swl/swl_wincore.cpp]: Why this condition was initially inside the loop?

Comment (helgp) [inline src/swl/swl_wincore.cpp]: Most likely, this was an error. 

Function name and its use cases clue that the function should traverse through tree hierarchy for all WCM windows, and fill the list by the window IDs, which have WT_POPUP attribute. It is clear that old code is not doing the task.

Comment (corporateshark) [inline src/swl/swl_wincore.cpp]: Ok, waiting for CI.

Comment (helgp) [inline src/swl/swl_wincore.cpp]: Tree hierarchy dump function (Win::dbg_dump()) has revealed that the hierarchy with popup windows has the structure:

RootWindow1 -> NonPopupWindow2 ->PopupWindow3

Original code stops traversing into child windows on the RootWindow1 level, and does not put the PopupWidow3 to the list.

**Pull request merged**